server/src/graphql/api.ts

```diff
@@ -24,22 +24,29 @@ interface Context {
 export const graphqlRoot: Resolvers<Context> = {
   Art: {
     creator: self => User.findOne({ id: self.creatorId }) as any,
+    views: self => Art.findOne({ id: self.id }).then(art => art!.views) as any,
+    likes: self => Art.findOne({ id: self.id }).then(art => art!.likes) as any,
+    seen: async (self, _, ctx) => {
+      if (!ctx.user) return null
+      const count = await getConnection()
+        .createQueryBuilder()
+        .select('1')
+        .from('art_views_user', 'avu')
+        .where('artId = :aid', { aid: self.id })
+        .andWhere('userId = :uid', { uid: ctx.user.id })
+        .getCount()
+      return count === 1
+    },
   },
   User: {
     artworkCreated: self => User.findOne({ where: { id: self.id } }).then(user => user!.artworkCreated) as any,
-    artSeen: self => User.findOne({ where: { id: self.id } }).then(user => user!.artSeen) as any,
-    artLiked: self => User.findOne({ where: { id: self.id } }).then(user => user!.artLiked) as any,
+    artSeen: self => User.findOne({ where: { id: self.id } }).then(user => user!.artworkSeen) as any,
+    artLiked: self => User.findOne({ where: { id: self.id } }).then(user => user!.artworkLiked) as any,
   },
   Query: {
     self: (_, args, ctx) => (ctx.user as any) || null,
-    art: async (_, { id }) => {
-      const thing = (await Art.findOne({ where: { id: id } })) || null
-      return thing as any
-    },
-    arts: async (_, args, ctx) => {
-      const response: Art[] = await Art.find({ take: 128 })
-      return response as any
-    },
+    art: async (_, { id }) => Art.findOne({ where: { id: id } }) as any,
+    arts: async () => Art.find({ take: 128 }) as any,
```

Comment:
Removed take: 128 on my branch

Reply:
I'm going to leave this in, since we never query arts on the frontend (as far as I can see), and this way debugging with a large number of arts will be efficient